util/importers/ldc93s1.py

```diff
@@ -0,0 +1,155 @@
+import wave
+import random
+import shutil
+import threading
+import numpy as np
+
+from os import path
+from os import rmdir
+from os import remove
+from glob import glob
+from math import ceil
+from Queue import Queue
+from os import makedirs
+from itertools import cycle
+from os.path import getsize
+from threading import Thread
+from Queue import PriorityQueue
+from util.gpu import get_available_gpus
+from util.text import texts_to_sparse_tensor
+from tensorflow.python.platform import gfile
+from util.audio import audiofile_to_input_vector
+from tensorflow.contrib.learn.python.learn.datasets import base
+
+class DataSets(object):
+    def __init__(self, train, dev, test):
+        self._dev = dev
+        self._test = test
+        self._train = train
+
+    @property
+    def train(self):
+        return self._train
+
+    @property
+    def dev(self):
+        return self._dev
+
+    @property
+    def test(self):
+        return self._test
+
+class DataSet(object):
+    def __init__(self, graph, txt_files, thread_count, batch_size, numcep, numcontext):
+        self._graph = graph
+        self._numcep = numcep
+        self._batch_queue = Queue(2 * self._get_device_count())
+        self._txt_files = txt_files
+        self._batch_size = batch_size
+        self._numcontext = numcontext
+        self._thread_count = thread_count
+        self._files_circular_list = self._create_files_circular_list()
+        self._start_queue_threads()
+
+    def _get_device_count(self):
+        available_gpus = get_available_gpus()
+        return max(len(available_gpus), 1)
+
+    def _start_queue_threads(self):
+        batch_threads = [Thread(target=self._populate_batch_queue) for i in xrange(self._thread_count)]
+        for batch_thread in batch_threads:
+            batch_thread.daemon = True
+            batch_thread.start()
+
+    def _create_files_circular_list(self):
+        priorityQueue = PriorityQueue()
+        for txt_file in self._txt_files:
+          wav_file = path.splitext(txt_file)[0] + ".wav"
+          wav_file_size = getsize(wav_file)
+          priorityQueue.put((wav_file_size, (txt_file, wav_file)))
+        files_list = []
+        while not priorityQueue.empty():
+            priority, (txt_file, wav_file) = priorityQueue.get()
+            files_list.append((txt_file, wav_file))
+        return cycle(files_list)
+
+    def _populate_batch_queue(self):
```

Comment:
That should be good and I could do something simpler, and that looks to be running as expected :)